teps/0002-custom-tasks.md

```diff
@@ -414,3 +414,13 @@ None.
 ## Upgrade & Migration Strategy (optional)
 
 TBD. At this time, the proposal only covers adding new a type and documentating the contract. If changes to the types or contract are deemed necessary in the future, in response to feedback, then an upgrade/migration strategy might be necessary.
+
+# Open Questions
+
+* Should Tekton's controller be responsible for updating `Run`s' `.status.conditions` in the case of cancellation and timeout (as it does when enforcing [initial update timeout](#initial-update-timeout)), or should these updates be the sole responsibility of Custom Task authors?
+
+* Package name and helper methods included in `tektoncd/pipeline` repo to aid Custom Task authors writing their controllers in Go; and should we expect them to use `knative/pkg` helpers too?
+
+* Versioning and release cadence and ownership of `tektoncd/sample-task` repo; will it be released/versioned alongside `tektoncd/pipeline`?
```

Comment:
is sample-task different from the package name and helper methods above?

i suggest we start releasing and versioning it separately to get a feel for when it actually needs updates and then decide based on what we see happening (if there is no dependency b/w the two, coupling them seems arbitrary)

Reply:
Yeah, there's two things I'd like us to provide:

1. A package of helper methods in `tektoncd/pipeline`, versioned with Tekton releases
1. A sample-task template repo that's forkable, and can give authors a good starting point to build new custom tasks. This would necessarily use some of the helpers in (1). It probably doesn't need to be versioned (authors can just fork from HEAD), but I think there's an open question about whether it should only depend on tagged/released `tektoncd/pipeline` code, or from HEAD (probably the former?)

Reply:
> A package of helper methods in tektoncd/pipeline, versioned with Tekton releases

when you say helpers, what kinda stuff do you have in mind?

Reply:
[`FilterRunRef`](https://github.com/ImJasonH/cel-task/blob/6c75268b6b49859f3680ceab0cbbf938191a9606/main.go#L50), which is used by a controller to request a filtered list of `Run` updates for only those that match their specified apiVersion+kind

And some reflect-based param-placeholder-replacing helper that would likely either reuse large chunks of existing Tekton code, or be so generic as to be directly usable as TaskRun/PipelineRun placeholder resolving directly.

Beyond that nothing jumps to mind, but we might come up with more in the course of writing more custom tasks.

Reply:
Interesting! Do you see the Pipelines controller using FilterRunRef? It seems kinda like it might only be used by the custom controller 🤔 

I'm actually starting to wonder if it will make sense in the long run to have the Run type and supporting libs like this together in the separate repo, imported by tekton pipelines

(not something we have to decide right away but just wondering)

Reply:
I don't see Tekton's own controllers using `FilterRunRef`, since if any part of Tekton cares about `Run`s (tbd, see other thread) then it will care about _all_ of them, and not just those of a given ref-type.

The main reason for putting it in the `tektoncd/pipeline` repo is to avoid repo proliferation and interdependency hell. We could have a `tektoncd/pkg` that only provides helper methods for custom task authors, but then it would basically just have these two methods, and since the param-replacement method would hopefully be used by `tektoncd/pipeline` itself, it would have to import that repo, which means it would have to be released/versioned in cadence with `tektoncd/pipeline`, and I figured ugh let's just put it in the main repo directly. Also setting up CI and OWNERS for this new repo, ugh.

FWIW the PR to add `FilterRunRef` to `tektoncd/pipeline` is here: https://github.com/tektoncd/pipeline/pull/2915